packages/derive/src/lib.rs

```diff
@@ -101,32 +119,76 @@ pub fn entry_point(
     entry_point_impl(attr.into(), item.into()).into()
 }
 
-fn entry_point_impl(attr: TokenStream, mut item: TokenStream) -> TokenStream {
-    let cloned = item.clone();
-    let function: syn::ItemFn = maybe!(syn::parse2(cloned));
+fn expand_attributes(func: &mut ItemFn) -> syn::Result<TokenStream> {
+    let mut attributes = std::mem::take(&mut func.attrs);
+    let mut stream = TokenStream::new();
+    while let Some(attribute) = attributes.pop() {
+        if !attribute.path().is_ident("set_contract_state_version") {
+            func.attrs.push(attribute);
```

Comment:
This reverses the order of attributes. Can that be a problem?

Reply:
Potentially could be since in some circumstances the order of attribute macro execution could be important (if the macros aren't executed before ours; I think execution is from outer macro to inner?).

I just changed the code. The weird `while let pop()` was an artifact from an older draft of the macro anyway where I wanted to avoid allocating another vector for the attributes we wanted to keep.